include/lbann/comm.hpp

```diff
@@ -1057,6 +1064,18 @@ class lbann_comm {
   using al_comms_key_type = std::pair<MPI_Comm, std::type_index>;
   using al_comms_val_type = std::unique_ptr<::Al::MPICommunicator>;
   std::map<al_comms_key_type, al_comms_val_type> m_al_comms;
+#ifdef AL_HAS_NCCL
+  /** Number of streams to round-robin between for NCCL allreduces. */
```

Comment:
We should add a note explaining why a prime number is preferable to a nice, round number.